tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedErrorMatcher.cs

```diff
@@ -0,0 +1,145 @@
+                        break;
+                    case "isClientError":
+                        AssertIsClientError(actualException, element.Value.AsBoolean);
+                        break;
+                    case "errorContains":
+                        AssertErrorContains(actualException, element.Value.AsString);
+                        break;
+                    case "errorCode":
+                        AssertErrorCode(actualException, element.Value.AsInt32);
+                        break;
+                    case "errorCodeName":
+                        AssertErrorCodeName(actualException, element.Value.AsString);
+                        break;
+                    case "errorLabelsContain":
+                        AssertErrorLabelsContain(actualException, element.Value.AsBsonArray.Select(x => x.AsString));
+                        break;
+                    case "errorLabelsOmit":
+                        AssertErrorLabelsOmit(actualException, element.Value.AsBsonArray.Select(x => x.AsString));
+                        break;
+                    case "expectResult":
+                        AssertExpectResult(actualException, element.Value.AsBsonDocument);
+                        break;
+                    default:
+                        throw new FormatException($"Unrecognized error assertion: '{element.Name}'");
+                }
+            }
+        }
+
+        private void AssertErrorCode(Exception actualException, int expectedErrorCode)
+        {
+            var mongoCommandException = actualException.Should().BeOfType<MongoCommandException>().Subject;
+
+            mongoCommandException.Code.Should().Be(expectedErrorCode);
+        }
+
+        private void AssertErrorCodeName(Exception actualException, string expectedErrorCodeName)
+        {
+            var mongoCommandException = actualException.Should().BeOfType<MongoCommandException>().Subject;
+
+            mongoCommandException.CodeName.Should().Be(expectedErrorCodeName);
+        }
+
+        private void AssertErrorContains(Exception actualException, string expectedSubstring)
+        {
+            actualException.Message.Should().ContainEquivalentOf(expectedSubstring);
+        }
+
+        private void AssertErrorLabelsContain(Exception actualException, IEnumerable<string> expectedErrorLabels)
+        {
+            actualException.Should().BeAssignableTo<MongoException>();
+            var mongoException = actualException as MongoException;
+
+            mongoException.ErrorLabels.Should().Contain(expectedErrorLabels);
+        }
+
+        private void AssertErrorLabelsOmit(Exception actualException, IEnumerable<string> expectedAbsentErrorLabels)
+        {
+            actualException.Should().BeAssignableTo<MongoException>();
+            var mongoException = actualException as MongoException;
+
+            mongoException.ErrorLabels.Should().NotContain(expectedAbsentErrorLabels);
```

Comment:
Replace 97-100 with:

            var mongoException = actualException.Should().BeAssignableTo<MongoException>().Subject;
            mongoException.ErrorLabels.Should().NotContain(expectedAbsentErrorLabels);